mod-source-record-storage-server/src/main/java/org/folio/dao/AbstractEntityDao.java

```diff
@@ -61,109 +64,126 @@ public void getByQuery(Q query, int offset, int limit, String tenantId, Handler<
     String where = query.toWhereClause();
     String orderBy = query.toOrderByClause();
     String sql = String.format(GET_BY_QUERY_SQL_TEMPLATE, getColumns(), getTableName(), where, orderBy, offset, limit);
-    log.info("Attempting stream get by query: {}", sql);
-    postgresClientFactory.createInstance(tenantId).getClient().getConnection(connection -> {
-      if (connection.failed()) {
-        log.error("Failed to get database connection", connection.cause());
-        endHandler.handle(Future.failedFuture(connection.cause()));
+    log.info("Attempting stream get by filter: {}", sql);
+    postgresClientFactory.getClient(tenantId).getConnection(ar1 -> {
+      if (ar1.failed()) {
+        log.error("Failed to get database connection", ar1.cause());
+        endHandler.handle(Future.failedFuture(ar1.cause()));
         return;
       }
-      connection.result().queryStream(sql, stream -> {
-        if (stream.failed()) {
-          log.error("Failed to get stream", stream.cause());
-          endHandler.handle(Future.failedFuture(stream.cause()));
+      SqlConnection connection = ar1.result();
+      connection.prepare(sql, ar2 -> {
+        if (ar2.failed()) {
+          log.error("Failed to prepare query", ar2.cause());
+          endHandler.handle(Future.failedFuture(ar2.cause()));
           return;
         }
-        stream.result()
+        PreparedStatement pq = ar2.result();
+        Transaction tx = connection.begin();
```

Comment:
Can we move transaction into the DaoUtil and create a template method?

Reply:
Done